Breeze/UserDefaults.swift

```diff
@@ -211,6 +229,100 @@ extension UserDefaults{
+            let currWeekUsage = integer(forKey: UserDefaultsKeys.currWeekPhoneUsage.rawValue)
+            
+            if (currDayOfWeek == 1) { //Sunday: roll over statistics
+                set(currWeekUsage, forKey: UserDefaultsKeys.prevWeekPhoneUsage.rawValue)
+                set(0, forKey: UserDefaultsKeys.currWeekPhoneUsage.rawValue)
+                rollOverNotifications()
+            }
+        }
+    }
+    
+    func getCurrNotificationSends() -> Int {
+        return integer(forKey: UserDefaultsKeys.notificationsSentCurrWeek.rawValue)
+    }
+    
+    func getCurrNotificationClicks() -> Int {
+        return integer(forKey: UserDefaultsKeys.notificationsClickedCurrWeek.rawValue)
+    }
+    
+    func getCurrNotificationSnoozes() -> Int {
+        return integer(forKey: UserDefaultsKeys.notificationsSnoozedCurrWeek.rawValue)
+    }
+    
+    func getLastWeekNotificationSends() -> Int {
+        return integer(forKey: UserDefaultsKeys.notificationsSentLastWeek.rawValue)
+    }
+    
+    func getLastWeekNotificationClicks() -> Int {
+        return integer(forKey: UserDefaultsKeys.notificationsClickedLastWeek.rawValue)
+    }
+    
+    func getLastWeekNotificationSnoozes() -> Int {
+        return integer(forKey: UserDefaultsKeys.notificationsSnoozedLastWeek.rawValue)
+    }
+    
+    func addNotificationSent() {
+        let currNotificationSends = getCurrNotificationSends()
+        set(currNotificationSends + 1, forKey: UserDefaultsKeys.notificationsSentCurrWeek.rawValue)
+    }
+    func addNotificationClick() {
+        let currNotificationClicks = getCurrNotificationClicks()
+        set(currNotificationClicks + 1, forKey: UserDefaultsKeys.notificationsClickedCurrWeek.rawValue)
+    }
+    
+    func addNotificationSnooze() {
+        let currNotificationSnoozes = getCurrNotificationSnoozes()
+        set(currNotificationSnoozes + 1, forKey: UserDefaultsKeys.notificationsSnoozedCurrWeek.rawValue)
+    }
+
+    func rollOverNotifications() {
+        let currNotificationSends = getCurrNotificationSends()
+        let currNotificationClicks = getCurrNotificationClicks()
+        let currNotificationSnoozes = getCurrNotificationSnoozes()
+        
+        set(currNotificationSends, forKey: UserDefaultsKeys.notificationsSentLastWeek.rawValue)
+        set(currNotificationClicks, forKey: UserDefaultsKeys.notificationsClickedLastWeek.rawValue)
+        set(currNotificationSnoozes, forKey: UserDefaultsKeys.notificationsSnoozedLastWeek.rawValue)
+    }
+    
+    func getProportionWeekSpent() -> Double {
+        //Current Date
+        let calendar = Calendar.current
+        let currDate = Date()
+        let dayComponents = calendar.dateComponents([.weekday], from: currDate)
+        let dayOfWeek = Int(dayComponents.weekday ?? 1)
+        let hourOfDay = calendar.component(.hour, from: currDate)
+        print("day: " + String(dayOfWeek))
+        print("hour: " + String(hourOfDay)) //HOW TO GET THIS TO MILITARY TIME
```

Comment:
I looked this up and found this thread: https://developer.apple.com/forums/thread/131724 , so I think we need to put this at the top of the method
let dateFormat = DateFormatter.dateFormat (fromTemplate: "j",options:0, locale: Locale.current)